### PR TITLE
Add dependency on ucs for pgf tests using [utf8x]{inputenc}.

### PR DESCRIPTION
**Pull request by anntzer**
`ucs` is a dependency of `\usepackage[utf8x]{inputenc}`, but if it
is absent we fail later with the slightly confusing error message
"Keyboard character used is undefined in inputencoding `utf8x'" (as
can be observed by temporarily uninstalling `ucs`).  Instead, make
the dependency explicit.  Also, in test_rcupdate, only one of the two
iterations uses `[utf8x]{inputenc}`, so run the other iteration first
(giving it a chance to fail) and skip the test only at the second
iteration if needed.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/tests/test_backend_pgf.py [hunk 1]
```diff
@@ -153,6 +153,10 @@ def test_rcupdate():
                                  '\\usepackage{sfmath}')}]
     tol = [6, 0]
     for i, rc_set in enumerate(rc_sets):
+        if i == 1:
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_backend_pgf.py:156]: Less magical:
```suggestion
        if 'utf8x' in rc_set.get('pgf.preamble', ''):
```

Comment (anntzer) [reply lib/matplotlib/tests/test_backend_pgf.py]: good idea, went for a variant of it.

File: lib/matplotlib/tests/test_backend_pgf.py [hunk 2]
```diff
@@ -113,6 +113,7 @@ def test_xelatex():
 
 # test compiling a figure to pdf with pdflatex
 @needs_pdflatex
+@pytest.mark.skipif(not _has_tex_package('ucs'), reason='needs ucs.sty')
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_backend_pgf.py:116]: did you mean to remove this line?

Comment (anntzer) [reply lib/matplotlib/tests/test_backend_pgf.py:116]: no?  that test also needs ucs (and it's a single figure test so we can just use the decorator form)

**Review by timhoffm — APPROVED** [review]

**Review by QuLogic — APPROVED** [review]

**Pull request merged**